### DISCUSSION TITLE
Symbol Collision: gstreamer depends on both GTK+3 and GTK4

### DISCUSSION THREAD

**Discussion by challinan** [Homebrew/discussions#4633] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.28
ORIGIN: https://github.com/Homebrew/brew
HEAD: 2b4cfb033757b63ab0b128d01c58db76480956f9
Last commit: 3 days ago
Core tap JSON: 09 Jul 16:59 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
$ brew doctor
Your system is ready to brew.
```

### Description of issue

Blank lines added to aid readability

It seems gstreamer has dependencies on both gtk+3 and gtk4, which apparently both supply certain classes.  Attempting to uninstall either gtk+3 or gtk4 complains that both are required by gstreamer.

`$$ brew deps gstreamer | grep gtk
gtk+3
gtk4'

`$ gst-device-monitor-1.0 Audio/Source`

`objc[39196]: Class ResultReceiver is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.38/lib/libgtk-3.0.dylib (0x106ea5df8) and /opt/homebrew/Cellar/gtk4/4.10.4/lib/libgtk-4.1.dylib (0x113eaaec0). One of the two will be used. Which one is undefined.`

`objc[39196]: Class GtkApplicationQuartzDelegate is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.38/lib/libgtk-3.0.dylib (0x106ea5e20) and /opt/homebrew/Cellar/gtk4/4.10.4/lib/libgtk-4.1.dylib (0x113eaaee8). One of the two will be used. Which one is undefined.`

`objc[39196]: Class GNSMenuItem is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.38/lib/libgtk-3.0.dylib (0x106ea5e70) and /opt/homebrew/Cellar/gtk4/4.10.4/lib/libgtk-4.1.dylib (0x113eaaf38). One of the two will be used. Which one is undefined.`

`objc[39196]: Class GNSMenu is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.38/lib/libgtk-3.0.dylib (0x106ea5e98) and /opt/homebrew/Cellar/gtk4/4.10.4/lib/libgtk-4.1.dylib (0x113eaaf60). One of the two will be used. Which one is undefined.`

`objc[39196]: Class FilterComboBox is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.38/lib/libgtk-3.0.dylib (0x106ea5f10) and /opt/homebrew/Cellar/gtk4/4.10.4/lib/libgtk-4.1.dylib (0x113eaafd8). One of the two will be used. Which one is undefined.`

`objc[39196]: Class gdkCoreCursor is implemented in both /opt/homebrew/Cellar/gtk+3/3.24.38/lib/libgdk-3.0.dylib (0x1061f9580) and /opt/homebrew/Cellar/gtk4/4.10.4/lib/libgtk-4.1.dylib (0x113eab028). One of the two will be used. Which one is undefined.`

**Comment by cbasavaraj** [comment]
I have the same problem, can't get even a simple gstreamer pipeline with `autovideosink` to run on my MacBook Pro M2. Any suggestions?
